loader/src/task_local.rs

```diff
@@ -0,0 +1,187 @@
+use pin_project_lite::pin_project;
+use std::cell::RefCell;
+use std::error::Error;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::{fmt, thread};
+
+// Reproduced from `tokio` (under MIT license) at https://github.com/tokio-rs/tokio/blob/9a3603fa75ff854e007d372061edf47cf8d02690/tokio/src/task/task_local.rs.
```

Comment:
Trying to get rid of the `task_local` seemed like a lot of work compared to just pulling it in as a single file dependency.

Reply:
I've no idea how it works, but glad it does!